js/ui/hierarchical_collection/ui.hierarchical_collection_widget.js

```diff
@@ -85,7 +89,7 @@ const HierarchicalCollectionWidget = CollectionWidget.inherit({
                     .append(this._getTextContainer(itemData))
                     .append(this._getPopoutContainer(itemData));
                 that._addContentClasses(itemData, $container.parent());
-            }.bind(this), ['text', 'html', 'items', 'icon'], this.option('integrationOptions.watchMethod'), {
+            }.bind(this), fields, this.option('integrationOptions.watchMethod'), {
```

Comment:
Seems like this code is not tested, am i right?

Reply:
If I revert that changes then all my test will broken, because extra fields like url and linkAttr will not be in attributes

Reply:
oh, ok